src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2886,55 +2936,124 @@ private BoundExpression BindIncrementOperator(ExpressionSyntax node, ExpressionS
+                string ordinaryName,
+                BoundExpression operand,
+                BindingDiagnosticBag diagnostics)
+            {
+                Debug.Assert(operand.Type is not null);
+                Debug.Assert(mode != InstanceUserDefinedIncrementUsageMode.None);
+
+                CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = GetNewCompoundUseSiteInfo(diagnostics);
+
+                ArrayBuilder<MethodSymbol>? methods = LookupUserDefinedInstanceOperators(
+                    operand.Type,
+                    checkedName: checkedName,
+                    ordinaryName: ordinaryName,
                     parameterCount: 0,
                     ref useSiteInfo);
 
+                diagnostics.Add(node, useSiteInfo);
+
                 if (methods?.IsEmpty != false)
                 {
-                    diagnostics.Add(node, useSiteInfo);
                     methods?.Free();
                     return null;
                 }
 
+                AnalyzedArguments? analyzedArguments = null;
+                BoundIncrementOperator? inPlaceResult = tryInstanceOperatorOverloadResolutionAndFreeMethods(node, operatorToken, kind, mode, isChecked, isExtension: false, operand, ref analyzedArguments, methods, diagnostics);
+                Debug.Assert(analyzedArguments is not null);
+                analyzedArguments.Free();
+
+                return inPlaceResult;
+            }
+
+            BoundIncrementOperator? tryInstanceOperatorOverloadResolutionAndFreeMethods(
+                ExpressionSyntax node,
+                SyntaxToken operatorToken,
+                UnaryOperatorKind kind,
+                InstanceUserDefinedIncrementUsageMode mode,
+                bool checkOverflowAtRuntime,
+                bool isExtension,
+                BoundExpression operand,
+                ref AnalyzedArguments? analyzedArguments,
+                ArrayBuilder<MethodSymbol> methods,
+                BindingDiagnosticBag diagnostics)
+            {
                 Debug.Assert(!methods.IsEmpty);
 
+                var operandType = operand.Type;
+                Debug.Assert(operandType is not null);
+
                 var overloadResolutionResult = OverloadResolutionResult<MethodSymbol>.GetInstance();
                 var typeArguments = ArrayBuilder<TypeWithAnnotations>.GetInstance();
-                var analyzedArguments = AnalyzedArguments.GetInstance();
+
+                if (analyzedArguments == null)
+                {
+                    analyzedArguments = AnalyzedArguments.GetInstance();
+
+                    if (isExtension)
+                    {
+                        // Create a set of arguments for overload resolution including the receiver.
+                        CombineExtensionMethodArguments(operand, originalArguments: null, analyzedArguments);
+
+                        if (operandType.IsValueType)
+                        {
+                            analyzedArguments.RefKinds.Add(RefKind.Ref);
```

Comment:
This is a bit different than how we handled the implicit `ref` on receiver in other scenarios. 
For method invocations and property access, we leave the argument information as `RefKind.None` and let downstream components (such as overload resolution) deal with it. See `resolveMethods`.
Could we do the same here?